packages/gafl-webapp-service/src/pages/contact/contact/result-function.js

```diff
@@ -1,10 +1,12 @@
-import { CONTACT_SUMMARY_SEEN, CommonResults } from '../../../constants.js'
+import { CONTACT_SUMMARY_SEEN, CommonResults, CHANGE_CONTACT_DETAILS_SEEN } from '../../../constants.js'
 
 export default async request => {
   const status = await request.cache().helpers.status.getCurrentPermission()
   const transaction = await request.cache().helpers.transaction.getCurrentPermission()
 
-  return transaction.isLicenceForYou === false || status.fromSummary === CONTACT_SUMMARY_SEEN || transaction.isRenewal
-    ? CommonResults.SUMMARY
-    : CommonResults.OK
+  return status.fromContactDetailsSeen === CHANGE_CONTACT_DETAILS_SEEN.SEEN
+    ? CommonResults.AMEND
+    : transaction.isLicenceForYou === false || status.fromSummary === CONTACT_SUMMARY_SEEN || transaction.isRenewal
+      ? CommonResults.SUMMARY
+      : CommonResults.OK
```

Comment:
Surprised Sonar Cloud didn't pick up on this - but ternaries in ternaries isn't a great look, please would you refactor to an `if...else if` structure to make it more readable? Also, the multi-conditions for SUMMARY / OK would probably be better taken out into a variable: `const showSummary = !transaction.isLicenceForYou || status.fromSummary === CONTACT_SUMMARY_SEEN || transaction.isRenewal`.

Finally, I noticed there don't seem to be any tests for your new CommonResults.AMEND condition, please will you add some?